### PR TITLE
Ends on scheduling recurrences

### PR DESCRIPTION
**Pull request by Angatupyry**
## What's new

- End date added for recurring tasks
- Ending date is inclusive

#### How test it?
- Create a scheduled task. 
- Choose the ON option under "Ends" and set an until date.
- The scheduling screen should show only tasks up to the chosen date. 

[chrome-capture-2023-6-21.webm](https://github.com/open-rmf/rmf-web/assets/37310205/9f265ad0-12c1-4181-8631-8230d1aa60e1)

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: packages/react-components/lib/tasks/create-task.tsx [hunk 1]
```diff
@@ -1246,6 +1277,51 @@ export function CreateTaskForm({
...
+                          return { ...prev, until: date };
+                        })
+                      }
+                      disabled={scheduleUntilValue !== ScheduleUntilValue.ON}
+                      renderInput={(props) => <TextField {...props} fullWidth />}
+                    />
+                  </Grid>
+                </Grid>
+              </RadioGroup>
+            </FormControl>
```
Comment (aaronchongth) [inline packages/react-components/lib/tasks/create-task.tsx:1323]:
```suggestion
           <FormControl fullWidth={true}>
              <FormHelperText>Ends</FormHelperText>
              <RadioGroup
                aria-labelledby="controlled-radio-buttons-group"
                name="controlled-radio-buttons-group"
                value={scheduleUntilValue}
                onChange={handleScheduleUntilValue}
                row
              >
                <Grid item xs={6} paddingLeft={theme.spacing(1)}>
                  <FormControlLabel
                    value={ScheduleUntilValue.NEVER}
                    control={<Radio />}
                    label="Never"
                  />
                </Grid>
                <Grid item xs={2} paddingLeft={theme.spacing(1)}>
                  <FormControlLabel
                    value={ScheduleUntilValue.ON}
                    control={<Radio />}
                    label="On"
                  />
                </Grid>
                <Grid item xs={4}>
                  <DatePicker
                    value={
                      scheduleUntilValue === ScheduleUntilValue.NEVER
                        ? new Date()
                        : schedule.until
                    }
                    onChange={(date) =>
                      date &&
                      setSchedule((prev) => {
                        date.setHours(23);
                        date.setMinutes(59);
                        return { ...prev, until: date };
                      })
                    }
                    disabled={scheduleUntilValue !== ScheduleUntilValue.ON}
                    renderInput={(props) => <TextField {...props} fullWidth />}
                  />
                </Grid>
              </RadioGroup>
            </FormControl>
```

I think we can use `FormHelperText` for the label, so it'll look similar to the labels of the other fields. My other suggestions here are mainly for using up empty space horizontally,

it'll look like this 
image

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Tested and working! This is awesome, thanks @Angatupyry!

I've left 1 review comment regarding the form, and how I would suggest shuffling things around and aligning items. Let me know what you think

File: packages/react-components/lib/tasks/create-task.tsx (hunk 1 above)
Comment (Angatupyry) [reply packages/react-components/lib/tasks/create-task.tsx]: This is great!
Thanks!
Done 8b8c8d85fc658518658664a1e610d8e0f72f3d00

**Review by aaronchongth — APPROVED** [review]
LGTM! Thanks for the quick fix!

**Pull request merged**